tests/parallel_runner.py

```diff
@@ -0,0 +1,160 @@
+    self.init_processes(test_queue)
+    results = self.collect_results()
+    return self.combine_results(result, results)
+
+  def create_test_queue(self):
+    test_queue = multiprocessing.Queue()
+    for test in self.reversed_tests():
+      test_queue.put(test)
+    return test_queue
+
+  def reversed_tests(self):
+    tests = []
+    for test in self:
+      tests.append(test)
+    tests.sort(key=lambda test: str(test))
+    return tests[::-1]
+
+  def init_processes(self, test_queue):
+    self.processes = []
+    self.result_queue = multiprocessing.Queue()
+    for i in xrange(num_cores()):
+      p = multiprocessing.Process(target=self.testing_thread,
+                                  args=(test_queue, self.result_queue))
+      p.start()
+      self.processes.append(p)
+
+  @staticmethod
+  def testing_thread(work_queue, result_queue):
+    for test in iter(lambda: get_from_queue(work_queue), None):
+      result = BufferedParallelTestResult()
+      try:
+        test(result)
+      except Exception as e:
+        result.addError(test, e)
+      result_queue.put(result)
+
+  def collect_results(self):
+    buffered_results = []
+    while len(self.processes) > 0:
+      res = get_from_queue(self.result_queue)
+      if res is not None:
+        buffered_results.append(res)
+      else:
+        self.clear_finished_processes()
+    return buffered_results
+
+  def clear_finished_processes(self):
+    self.processes = filter(lambda p: p.is_alive(), self.processes)
+
+  def combine_results(self, result, buffered_results):
+    print
+    print 'DONE: combining results on main thread'
+    print
+    for r in sorted(buffered_results, key=lambda res:str(res.test)):
+      r.updateResult(result)
+    return result
+
+
+class BufferedParallelTestResult(object):
+  def __init__(self):
+    self.kind = None
+    self.test = None
+    self.error = None
+
+  def updateResult(self, result):
+    result.startTest(self.test)
+
+    if self.kind == TestResultType.Success:
+      result.addSuccess(self.test)
+    elif self.kind == TestResultType.Failure:
+      result.addFailure(self.test, self.error)
+    elif self.kind == TestResultType.Error:
+      result.addError(self.test, self.error)
+
+    result.stopTest(self.test)
+
+  def startTest(self, test):
+    pass
+  def stopTest(self, test):
+    pass
+
+  def addSuccess(self, test):
+    print test, '... ok'
+    self.kind = TestResultType.Success
+    self.test = test
+
+  def addFailure(self, test, err):
+    print test, '... FAIL:'
+    self.kind = TestResultType.Failure
+    self.test = test
+    self._set_error(err)
+
+  def addError(self, test, err):
+    print test, '... ERROR:'
+    self.kind = TestResultType.Error
+    self.test = test
+    self._set_error(err)
+
+  def _set_error(self, err):
+    a, b, tb = err
+    self.error = (a, b, FakeTraceback(tb))
+
+
+class TestResultType(enum.Enum):
+  Success = 1
+  Failure = 2
+  Error = 3
+
+
+class FakeTraceback(object):
```

Comment:
I'd just put a heredoc comment at the top of the class. And yeah that comment sounds about right. You could add that tb is a real traceback object that you are trying to mock.